content/assignments/memory-allocator/tests/ref/test-all.ref

```diff
@@ -0,0 +1,300 @@
+os_malloc (['0'])                                                                         = 0
+os_free (['0'])                                                                           = <void>
+os_malloc (['1000'])                                                                      = HeapStart + 0x18
+  brk (['0'])                                                                             = HeapStart + 0x0
+  brk (['HeapStart + 0x20000'])                                                           = HeapStart + 0x20000
+os_realloc (['HeapStart + 0x18', '2000'])                                                 = HeapStart + 0x18
+os_free (['HeapStart + 0x18'])                                                            = <void>
+os_calloc (['0', '100'])                                                                  = 0
+os_free (['0'])                                                                           = <void>
+os_calloc (['10', '100'])                                                                 = HeapStart + 0x18
+os_free (['HeapStart + 0x18'])                                                            = <void>
+os_calloc (['10', '100'])                                                                 = HeapStart + 0x18
+os_realloc (['HeapStart + 0x18', '132072'])                                               = <mapped-addr1> + 0x18
+  mmap (['0', '132096', 'PROT_READ | PROT_WRITE', 'MAP_PRIVATE | MAP_ANON', '-1', '0'])   = <mapped-addr1>
+os_free (['<mapped-addr1> + 0x18'])                                                       = <void>
+  munmap (['<mapped-addr1>', '132096'])                                                   = 0
+os_calloc (['1', '5000'])                                                                 = <mapped-addr2> + 0x18
+  mmap (['0', '5024', 'PROT_READ | PROT_WRITE', 'MAP_PRIVATE | MAP_ANON', '-1', '0'])     = <mapped-addr2>
+os_realloc (['<mapped-addr2> + 0x18', '2000'])                                            = HeapStart + 0x18
+  munmap (['<mapped-addr2>', '5024'])                                                     = 0
+os_realloc (['HeapStart + 0x18', '5000'])                                                 = HeapStart + 0x18
+os_free (['HeapStart + 0x18'])                                                            = <void>
+os_calloc (['1', '10'])                                                                   = HeapStart + 0x18
+os_malloc (['4023'])                                                                      = HeapStart + 0x40
+os_calloc (['1934', '1'])                                                                 = HeapStart + 0x1010
+os_calloc (['1', '80'])                                                                   = HeapStart + 0x17b8
+os_malloc (['653'])                                                                       = HeapStart + 0x1820
```

Comment:
The block that start at `0x1e50` will be chosen by the `find best` strategy since it has the size **160**, instead of the block that you described (`0x1980`) which has size **280**.
The `0x1e50` block cannot coalesce with anything since it is between 2 allocated blocks.